classes/services/queryBuilders/PKPSubmissionFileQueryBuilder.inc.php

```diff
@@ -0,0 +1,251 @@
+ * @file classes/services/QueryBuilders/PKPSubmissionFileQueryBuilder.php
+ *
+ * Copyright (c) 2014-2019 Simon Fraser University
+ * Copyright (c) 2000-2019 John Willinsky
+ * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+ *
+ * @class PKPSubmissionFileQueryBuilder
+ * @ingroup query_builders
+ *
+ * @brief Class for building database queries for submission files
+ */
+
+namespace PKP\Services\QueryBuilders;
+
+use Illuminate\Database\Capsule\Manager as Capsule;
+use PKP\Services\QueryBuilders\Interfaces\EntityQueryBuilderInterface;
+
+import('lib.pkp.classes.submission.SubmissionFile'); // SUBMISSION_FILE_ constants
+
+class PKPSubmissionFileQueryBuilder implements EntityQueryBuilderInterface {
+
+	/** @var array get submission files for one or more file stages */
+	protected $fileStages = [];
+
+	/** @var array get submission files for one or more genres */
+	protected $genreIds = [];
+
+	/** @var array get submission files for one or more review rounds */
+	protected $reviewRoundIds = [];
+
+	/** @var array get submission files for one or more review assignments */
+	protected $reviewIds = [];
+
+	/** @var array get submission files for one or more submissions */
+	protected $submissionIds = [];
+
+	/** @var array get submission files matching one or more files */
+	protected $fileIds = [];
+
+	/** @var array get submission files matching one or more ASSOC_TYPE */
+	protected $assocTypes = [];
+
+	/** @var array get submission files matching an ASSOC_ID with one of the assocTypes */
+	protected $assocIds = [];
+
+	/** @var boolean include submission files in the SUBMISSION_FILE_DEPENDENT stage */
+	protected $includeDependentFiles = false;
+
+	/**
+	 * Set fileStages filter
+	 *
+	 * @param array|int $fileStages
+	 * @return \PKP\Services\QueryBuilders\PKPSubmissionFileQueryBuilder
+	 */
+	public function filterByFileStages($fileStages) {
+		$this->fileStages = is_array($fileStages) ? $fileStages : [$fileStages];
+		return $this;
+	}
+
+	/**
+	 * Set genreIds filter
+	 *
+	 * @param array|int $genreIds
+	 * @return \PKP\Services\QueryBuilders\PKPSubmissionFileQueryBuilder
+	 */
+	public function filterByGenreIds($genreIds) {
+		$this->genreIds = is_array($genreIds) ? $genreIds : [$genreIds];
+		return $this;
+	}
+
+	/**
+	 * Set review rounds filter
+	 *
+	 * @param array|int $reviewRoundIds
+	 * @return \PKP\Services\QueryBuilders\PKPSubmissionFileQueryBuilder
+	 */
+	public function filterByReviewRoundIds($reviewRoundIds) {
+		$this->reviewRoundIds = is_array($reviewRoundIds) ? $reviewRoundIds : [$reviewRoundIds];
+		return $this;
+	}
+
+	/**
+	 * Set review assignments filter
+	 *
+	 * @param array|int $reviewIds
+	 * @return \PKP\Services\QueryBuilders\PKPSubmissionFileQueryBuilder
+	 */
+	public function filterByReviewIds($reviewIds) {
+		$this->reviewIds = is_array($reviewIds) ? $reviewIds : [$reviewIds];
+		return $this;
+	}
+
+	/**
+	 * Set submissionIds filter
+	 *
+	 * @param array|int $submissionIds
```

Comment:
I'm on the fence about accepting mixed types like this. Modern PHP array syntax makes `[$submissionId]` just as readable as `$submissionId` in calling code, and I don't love peppering functions like these with conditions to cast content into arrays. Open to ideas here but I'm trending against...

Reply:
You're right. The benefit isn't worth it.